modules/imgcodecs/test/test_gif.cpp

```diff
@@ -0,0 +1,204 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html
+#include "test_precomp.hpp"
+
+namespace opencv_test { namespace {
+TEST(Imgcodecs_Gif, read_gif_multi)
+{
+    const string root = cvtest::TS::ptr()->get_data_path();
+    const string filename = root+"gifsuite/imreadmulti_test/output.gif";
+    vector<cv::Mat> img_vec;
+    cv::imreadmulti(filename, img_vec,0,161);
+    const long unsigned int expected_size=161;
+    EXPECT_EQ(img_vec.size(), expected_size);
+    for(long unsigned int i=0;i<img_vec.size();i++){
+        cv::Mat img=img_vec[i];
+        const string xml_filename=root+"gifsuite/imreadmulti_test/frame"+std::to_string(i+1)+".xml";
+        FileStorage fs(xml_filename, FileStorage::READ);
+        EXPECT_TRUE(fs.isOpened());
+        Mat gt;
+        fs.getFirstTopLevelNode() >> gt;
+        EXPECT_PRED_FORMAT2(cvtest::MatComparator(0, 0), img, gt);
+    }
+}
+
+typedef testing::TestWithParam<string> Imgcodecs_Gif_GifSuite_SingleFrame;
+TEST_P(Imgcodecs_Gif_GifSuite_SingleFrame,read_gif_single)
+{
+    const string root = cvtest::TS::ptr()->get_data_path();
+    const string filename = root + "gifsuite/" + GetParam() + ".gif";
+    const string xml_filename =  root + "gifsuite/" + GetParam() + ".xml";
```

Comment:
The solution is very inefficient in terms of OpenCV Extra repository. I propose the following:
- Cherry pick several key images for regular unit testing, convert the reference xmls to png and put pngs to extra
- Add disabled test with all cases and describe how to get/generate data for it in the test comments. As alternative, you can put it to https://github.com/opencv/opencv_benchmarks as dedicated app with instruction

Reply:
here the testing data is converted from the testing images under `pngsuite` in extra via ffmpeg, thus we directly reference to the xmls that the png test uses. would this solution be fine?